### PR TITLE
Fix `VisualAI` clip feature extraction numpy() grad error

### PR DESCRIPTION
**Pull request by RizwanMunawar**
## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Safely detaches CLIP embeddings before converting to NumPy in the similarity search solution to prevent gradient-related issues and reduce memory overhead. 🔧

### 📊 Key Changes
- Added `.detach()` before `.cpu().numpy()` in:
  - `extract_image_feature`: `encode_image(...).detach().cpu().numpy()`
  - `extract_text_feature`: `encode_text(...).detach().cpu().numpy()`

### 🎯 Purpose & Impact
- Prevents PyTorch errors when converting tensors that require gradients to NumPy (common in non-training/inference flows). ✅
- Reduces unnecessary gradient tracking and memory usage during embedding extraction. 🧠💾
- Improves stability and compatibility across environments where autograd may be enabled. 🌐
- Non-breaking change; behavior remains the same for users, with fewer warnings/errors. 👍

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @username, thank you for submitting an `ultralytics/ultralytics` 🚀 PR! This is an automated message and an engineer will assist you shortly. To ensure a seamless integration of your work, please review the following checklist:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

**Review by UltralyticsAssistant — APPROVED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)</sub>

Change clean—detaching tensors before converting to NumPy is a safe fix to avoid gradient tracking during VisualAI feature extraction.

**Comment by RizwanMunawar** [comment]
## ⚡ Actions Trigger

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)<sub>

GitHub Actions below triggered via workflow dispatch on this PR branch `patch-fix-similarity-search` at 2025-10-22 08:34:34 UTC with `@ultralytics/run-ci` command:

* ✅ [CI](https://github.com/ultralytics/ultralytics/actions/runs/18710297656): `ci.yml` run 13474
* ✅ [Publish Docker Images](https://github.com/ultralytics/ultralytics/actions/runs/18710297796): `docker.yml` run 1323

**Comment by glenn-jocher** [comment]
@RizwanMunawar good fix! Dockers are all passing now :)

**Comment by UltralyticsAssistant** [comment]
🎉 Merge celebration! This refinement to our CLIP similarity search is a quiet power-up with big impact. As Benjamin Franklin said, “Well done is better than well said.” This precise `.detach()` addition turns potential autograd errors into smooth, memory-efficient inference—exactly the kind of craftsmanship that keeps our ecosystem robust.

- Thank you, @RizwanMunawar, for the clean, thoughtful fix.
- Appreciation to @glenn-jocher for the collaboration and review.

Example for clarity:
```python
image_feat = model.encode_image(img).detach().cpu().numpy()
text_feat = model.encode_text(txt).detach().cpu().numpy()
```

Merged and already making developers’ lives easier. Thank you for strengthening stability and performance across environments. See the details in PR [Improve CLIP embedding conversion — PR #22463](https://github.com/ultralytics/ultralytics/pull/22463). 🚀
